src/_pytest/config/__init__.py

```diff
@@ -1605,8 +1607,10 @@ def getini(self, name: str):
 
     # Meant for easy monkeypatching by legacypath plugin.
     # Can be inlined back (with no cover removed) once legacypath is gone.
-    def _getini_unknown_type(self, name: str, type: str, value: str | list[str]):
-        msg = f"unknown configuration type: {type}"
+    def _getini_unknown_type(self, name: str, type: str, value: object):
+        msg = (
+            f"Option {name} has unknown configuration type {type} with value {value!r}"
+        )
         raise ValueError(msg, value)  # pragma: no cover
```

Comment:
Could probably drop the second argument here now:

```suggestion
        raise ValueError(msg)  # pragma: no cover
```

as that's nicely in the text, and it looks like Python just raises that with a tuple as a message:

```python
>>> raise ValueError("test", 42)
Traceback (most recent call last):
  [...]
ValueError: ('test', 42)
```

But approving the PR as this shouldn't block it.

Reply:
Good catch. 👍